pyrobosim/pyrobosim/core/robot.py

```diff
@@ -284,8 +286,10 @@ def place_object(self, pose=None):
 
         # Validate the robot location
         loc = self.location
+        if isinstance(loc, str):
+            loc = self.world.get_entity_by_name(self.location)
         if not isinstance(loc, ObjectSpawn):
-            warnings.warn("Not an object spawn. Cannot place object.")
+            warnings.warn(f"{loc.name} is not an object spawn. Cannot place object.")
```

Comment:
Careful! If `loc` is something else (namely, `None`), this will error out... which I think was the main reason this wasn't in.

Maybe you can do an `elif ifinstance(loc, Location)` and then a catch-all `else` after?

Reply:
I think we can go with `loc` instead of `loc.name` and that should handle things well since the entities have  a `__repr__` and it will print `None` if it is `NoneType`